File: Interceptor/Java/src/main/java/me/hajoo/java/controller/HelloController.java
package me.hajoo.java.controller;

import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@Slf4j
public class HelloController {

    @GetMapping("/")
    public String hello() {
        log.info("/ 진입");
        return "Hello";
    }

    @GetMapping("/second-interceptor")
    public String hello2() {
        log.info("/second-interceptor 진입");
        return "Hello";
    }

    @GetMapping("/third/interceptor")
    public String hello3() {
        log.info("/third-interceptor 진입");
        return "Hello";
    }

}

File: MDC/src/main/kotlin/me/hajoo/mdc/controller/HelloController.kt
package me.hajoo.mdc.controller

import me.hajoo.mdc.common.logger
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController {

    companion object {
        const val HELLO_WORLD = "Hello World!"
    }

    @GetMapping
    fun hello() {
        logger().info(HELLO_WORLD)
        logger().warn(HELLO_WORLD)
        logger().error(HELLO_WORLD)
        logger().debug(HELLO_WORLD)
        logger().trace(HELLO_WORLD)
    }
}